src/viam/app/data/client.py

```diff
@@ -44,26 +57,109 @@ async def tabular_data_by_filter(
+        response: BinaryDataByFilterResponse = await self._data_client.BinaryDataByFilter(request, metadata=self._metadata)
+        if dest:
+            try:
+                file = open(dest, 'w')
+                file.write(f"{response.data}")
+            except Exception as e:
+                LOGGER.error(f"Failed to write binary data to file {dest}", exc_info=e)
+        return response.data
+
+    async def binary_data_by_ids(self, file_ids: Optional[List[str]], dest: Optional[str]) -> List[bytes]:
+        """Filter and download binary data
+
+        Args:
+            file_ids (List[str]): When supplied, limits binary data to the provided file IDs. If not provided, all data will be returned.
+            dest (str): When supplied, the binary data will be saved to the provided file path
+
+        Returns:
+            List[bytes]: The binary data
+        """
+        file_ids = file_ids if file_ids else []
+        request = BinaryDataByIDsRequest(file_ids=file_ids, include_binary=True)
+        response: BinaryDataByIDsResponse = await self._data_client.BinaryDataByIDs(request, metadata=self._metadata)
+        if dest:
+            try:
+                file = open(dest, 'w')
+                file.write(f"{response.data}")
+            except Exception as e:
+                LOGGER.error(f"Failed to write binary data to file {dest}", exc_info=e)
+        return response.data
 
     async def delete_tabular_data_by_filter(self, filter: Optional[Filter]) -> None:
-        raise NotImplementedError()
+        """Delete tabular data
+
+        Args:
+            filter (viam.app.data.Filter): When supplied, the tabular data to delete will be filtered based on the provided constraints.
+                If not provided, all data will be deleted. Exercise caution before using this option.
+        """
+        filter = filter if filter else Filter()
+        request = DeleteTabularDataByFilterRequest(filter=filter)
+        response: DeleteTabularDataByFilterResponse = await self._data_client.DeleteTabularDataByFilter(request, metadata=self._metadata)
 
     async def delete_binary_data_by_filter(self, filter: Optional[Filter]) -> None:
-        raise NotImplementedError()
+        """Delete binary data
 
-    async def delete_binary_data_by_ids(self, file_ids: Optional[List[str]]) -> None:
-        raise NotImplementedError()
+        Args:
+            filter (viam.app.data.Filter): When supplied, the binary data to delete will be filtered based on the provided constraints.
+                If not provided, all data will be deleted. Exercise caution before using this option.
+        """
+        filter = filter if filter else Filter()
+        request = DeleteBinaryDataByFilterRequest(filter=filter)
+        response: DeleteBinaryDataByFilterResponse = await self._data_client.DeleteBinaryDataByFilter(request, metadata=self._metadata)
+
+    async def delete_binary_data_by_ids(self, file_ids: List[str]) -> None:
```

Comment:
Still need to test this method but didn't want to block starting review, working with data team on getting a test set up.